### ISSUE TITLE
Editor loses focus after keyboard shortcut to switch to code editor

### ISSUE THREAD

**Issue by MatzeKitt** [WordPress/gutenberg#42145]
### Description

If I swith to the code editor via keyboard shortcut, I cannot switch right back afterwards via keyboard shortcut, since the editor loses the focus. I need to click inside the code editor to switch back to the visual editor.

If the more menu is open on the upper right corner of the editor, I can switch between both editors with keyboard shortcuts every time.

### Step-by-step reproduction instructions

1. Open a post/page edit page in the backend.
2. Click into the editor to get focus.
3. Press the keyboard shortcut to switch to the code editor.
4. Press the same keyboard shortcut again.

### Environment info

WordPress 6.0
No Gutenberg plugin installed
Safari 15.5
macOS 12.4

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by Mamaduka** [comment]
Probably related #40577.

**Comment by afercia** [comment]
Adding some more context from https://github.com/WordPress/gutenberg/issues/52719 (closed as duplicate).

Actually there are two problems with this:

1. For keyboard users and assistive technology users a focus loss is a terrible experience as they are 'catapulted' back to the document root and they have to start navigating the page from scratch.
2. When a full focus loss occurs, all the Editor keyboard shortcuts don't work any longer. For example Cmd+K to open the command center doesn't do anything. See https://github.com/WordPress/gutenberg/issues/52301

Please see https://github.com/WordPress/gutenberg/issues/52719 for some additional considerations.

**Comment by Mamaduka** [comment]
When/if code highlighting of the currently selected block is implemented (#26738), we can use it as a new target for the focus.

At the moment, I think @afercia's suggestion from https://github.com/WordPress/gutenberg/issues/52719#issuecomment-1639714329 could work as a middle ground.

**Comment by Mamaduka** [comment]
It looks like the keyboard shortcut now works as expected, even after the focus is moved to the document root.

**Comment by afercia** [comment]
Just tested on trunk and can confirm @Mamaduka feedback. I _think_ these shortcuts are now attached to a higher level of the React tree and work even when focus is on the document root.

However, when swithinc visual / code and vice-versa, the focus loss still occurs. As such, it appears  first part of the problem reported on https://github.com/WordPress/gutenberg/issues/52719 is now solved but focus still needs to be managed.
